mlos_bench/mlos_bench/storage/storage_factory.py

```diff
@@ -0,0 +1,52 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""
+Factory method to create a new Storage instance from configs.
```

Comment:
Can this replace `_load_storage` in launcher?

It would help simplify some of the `_load` and `build_generic` code which is really only used for `Storage` and `Optimizer` classes.

Reply:
uh, that'd be tough. The Launcher ties together a lot of disjoint components, and it is hard to split it into more or less independent parts. Because of that, I had to duplicate some of its functionality in the storage factory ☹️ 
I'll think of a better design later, but for now let's just keep the storage factory as is for the demo and make sure it looks good on the user side